### PR TITLE
Add Python 3.12 / Remove Python 3.8

### PR DESCRIPTION
**Pull request by mistafunk**
This includes an update of PyBind11 as well as a number of fixes and improvements in the Dockerfiles and CI pipeline scripts.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -283,7 +283,15 @@ def taskBuildDoc(cfg) {
...
-	String buildCmd = "python -m pip install . && sphinx-build docs ${cfg.ws}/${sphinxOutput}"
+    String venv = "${cfg.ws}/venv"
+	String buildCmd = ''
+	if (isUnix()) {
+        buildCmd += "cp -r /tmp/pyprt-build-venv ${venv}" // copy ro venv so we can install and build docs
+	    buildCmd += " && ${venv}/bin/python -m pip install . && ${venv}/bin/sphinx-build docs ${cfg.ws}/${sphinxOutput}"
+	}
+	else {
+        buildCmd = "python -m pip install . && sphinx-build docs ${cfg.ws}/${sphinxOutput}"
+	}
```
Comment (bneukom) [inline Jenkinsfile:294]: Minor formatting issues

File: Jenkinsfile [hunk 2]
```diff
@@ -301,7 +309,15 @@ def taskRunTests(cfg) {
...
 
-	String buildCmd = "python -m pip install . && python tests/run_tests.py --xml_output_directory ${cfg.ws}"
+    String venv = "${cfg.ws}/venv"
+	String buildCmd = ''
+	if (isUnix()) {
+	    buildCmd = "cp -r /tmp/pyprt-build-venv ${venv}" // copy ro venv so we can install to run tests
+	    buildCmd += " && ${venv}/bin/python -m pip install . && ${venv}/bin/python tests/run_tests.py --xml_output_directory ${cfg.ws}"
+	}
+	else {
+	    buildCmd = "python -m pip install . && python tests/run_tests.py --xml_output_directory ${cfg.ws}"
```
Comment (bneukom) [inline Jenkinsfile:319]: There seems to be a bit of code duplication in `taskBuildWheel/taskBuildConda/taskRunTests`. Could maybe be improved.

**Review by bneukom — APPROVED** [review]
Looks good from what I can tell!

File: Jenkinsfile (hunk 2 above)
Comment (mistafunk) [reply Jenkinsfile:319]: True, though it is always slightly different... not sure it is easier to read after pulling out stuff... will leave it like this for now, sry.

**Pull request merged**